pytorch_lightning/callbacks/progress.py

```diff
@@ -20,18 +20,38 @@
 """
 import importlib
 import sys
+from numbers import Number
 
 # check if ipywidgets is installed before importing tqdm.auto
 # to ensure it won't fail and a progress bar is displayed
 from typing import Optional, Union
 
 if importlib.util.find_spec('ipywidgets') is not None:
-    from tqdm.auto import tqdm
+    from tqdm.auto import tqdm as _tqdm
 else:
-    from tqdm import tqdm
+    from tqdm import tqdm as _tqdm
 
 from pytorch_lightning.callbacks import Callback
 
+_PAD_SIZE = 5
+
+
+class tqdm(_tqdm):
+    """ Custom tqdm progressbar where we append 0 to floating points/strings to
+        prevent the progress bar from flickering
+    """
+    @staticmethod
+    def format_num(n: Number) -> str:
+        """ Add additional padding to the formatted numbers """
+        should_be_padded = isinstance(n, (float, str))
+        if not isinstance(n, str):
+            n = _tqdm.format_num(n)
```

Comment:
```suggestion
            n = super().format_num(n)
```

Reply:
since it is a staticmethod, calling `super().format_num(n)` will fail with
`TypeError: super(type, obj): obj must be an instance or subtype of type`
If we still want to call super, i guess we need to do `super(_tqdm, _tqdm).format_num(n)`